test/shared/src/main/scala/zio/test/laws/GenF.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test.laws
+
+import zio.test.Gen
+
+trait GenF[-R, F[_]] {
+  def apply[R1 <: R, A](gen: Gen[R1, A]): Gen[R1, F[A]]
```

Comment:
This is really nice!

Reply:
Possibly this is another place where we might need to propagate the instances required of `A`.

Naively:

```scala
trait GenF[-R, F[_], Caps[_]] {
   def apply[R1 <: R, A: Caps](gen: Gen[R1, A]): Gen[R1, F[A]]
```

I'm thinking of the situations where, in order to create `F[A]`, the `A` itself must satisfy some properties. 

Or maybe there's a simpler way.